nbs/explains/background_tasks.ipynb

```diff
@@ -0,0 +1,296 @@
+    "    task = BackgroundTask(task_submit, request_time=request_time) # <7>\n",
+    "    return FtResponse(resp_content, background=task) # <8>\n",
+    "\n",
+    "@rt\n",
+    "def show_tstamps(): return Ul(*[Li(t) for t in tstamps()]) # <9> \n",
+    "\n",
+    "@rt\n",
+    "def index():\n",
+    "    return Titled('Background Task Dashboard',\n",
+    "        P(Button('Press to call slow service', # <10> \n",
+    "            hx_post='/submit', hx_target='#res')),\n",
+    "        H2('Responses from Tasks'),\n",
+    "        P('', id='res'),\n",
+    "        Div(Ul(*[Li(t) for t in tstamps()]),\n",
+    "            hx_get=show_tstamps, hx_trigger='every 5s'), # <11>\n",
+    "    )\n",
+    "\n",
+    "serve()\n",
+    "```\n",
+    "\n",
+    "1. Tracks when requests are sent and responses received\n",
+    "2. When the request was initiated\n",
+    "3. When the response was received\n",
+    "4. Task function calling slow service to be run in the background of a route handler. It is common but not necessary to prefix task functions with 'task_'\n",
+    "5. Call the slow API service (simulating a time-consuming operation)\n",
+    "6. Store both timestamps in our database\n",
+    "7. Create a background task by passing in the function to a BackgroundTask object, followed by any arguments.\n",
+    "8. In FtResponse, use the background keyword argument to set the task to be run after the HTTP response is generated.\n",
+    "9. Endpoint that displays all recorded timestamp pairs.\n",
+    "10. When this button is pressed, the 'submit' handler will respond instantly. The task_submit function will insert the slow API response into the db later. \n",
+    "11. Every 5 seconds get the tstamps stored in the DB."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "::: {.callout-tip}\n",
+    "\n",
+    "In the example above we use a synchronous background task function set in the `FtResponse` of a synchronous handler. However, we can also use asynchronous functions and handlers.\n",
+    "\n",
+    ":::"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Multiple background tasks in a handler"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "It is possible to add multiple background tasks to an FtResponse."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "::: {.callout-warning}\n",
+    "Multiple background tasks on a background task are executed in order. In the case a task raises an exception, following tasks will not get the opportunity to be executed.\n",
+    ":::"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```python\n",
+    "from fasthtml.common import *\n",
+    "from starlette.background import BackgroundTasks\n",
+    "\n",
+    "@rt\n",
+    "async def signup(email, username):\n",
```

Comment:
I don't think you need all this - IIRC you can just return multiple bg tasks in your returned tuple.